src/utils/fetchSetIndexComponents.ts

```diff
@@ -0,0 +1,32 @@
+import { camelCase } from 'lodash'
+
+const proxyPrefixURL = 'https://cors-anywhere.herokuapp.com/'
+const baseURL = 'https://api.tokensets.com/public/v2/portfolios/'
+const apiURL = proxyPrefixURL + baseURL
+
+export default (index: string) => {
+  return fetch(apiURL + index)
+    .then((response) => response.json())
+    .then((response) => {
+      if (!response?.portfolio?.components) {
+        // undocumented API endpoint. Throw error if not expected response format
+        throw new Error('Invalid API response from Set Protocol service')
+      }
+      const {
+        portfolio: { components },
+      } = response
+      const formattedComponents = components.map((component: any) => {
```

Comment:
Can you add a note here on what this formatting is for?
Or break up the formatting logic into a named helper function

Reply:
Yea sure. Right now it's only camelCasing strings but all the values are strings so wasn't sure if we want to format them to numbers here too. We use that `numeral` function on almost every input anyway so didn't seem necessary to.